src/types.ts

```diff
@@ -11,7 +11,7 @@ export type Adapter<T extends z.AnyZodObject = z.AnyZodObject> = {
   /**
    * Read the config
    */
-  read: () => Promise<z.infer<T>>;
+  read: () => Promise<z.infer<T>> | z.infer<T>;
```

Comment:
One small improvement, could be extracted to a separate PR. This basically allows any adapter to also return synchronous stuff.

Reply:
Yep, this is something that I wanted to introduce as well, can we please move it out from this PR, feel free to open a new one later just to introduce sync load config 🙏

Reply:
Done!